design/authenticators-status/authn_status_general.md

```diff
@@ -0,0 +1,113 @@
+# Feature Overview
+
+***As a*** Conjur operator\
+***I'd like to*** know that an authenticator is configured correctly\
+***So that*** I can complete the configuration properly if it's incomplete (or remove
+it if it's not needed).
+
+This feature lets the person who configures an authenticator to get an immediate feedback
+on the configuration, before any user needs to run an authentication request.
+
+# Implementation Details
+
+![alt text](authn-status-flow.jpeg "Authenticator Status API flow")
+
+We will create a new route in `routes.rb` for `/authenticators/:authenticator(/:service_id)/:account/status`.
+This route will lead to `AuthenticateController` which will consist of a new `status` method.
+In this method we will call a new CommandClass `Authentication::Status`. This class's
+`call` method will perform the general checks that apply to all authenticators
+and those specific to each authenticator:
+
+- The requesting user has access to the authenticator status route webservice
+    - more info in the Security section of the [epic](https://github.com/cyberark/conjur/issues/1062)
+- The authenticator is implemented
+- The account exists
+    - although a wrong account in an `/authenticate` request will indicate a client error and
+    not a server error, we still need to verify it exists in order to verify that the webservice exists
+    under that account
+- The webservice exists
+- The authenticator is enabled in the ENV
```

Comment:
@orenbm the authenticator is configured on the master, but the enabled authenticators environment variable is set for each an every Conjur instance. Will we tell the user that after configuring the authenticator on the master, checking the status should be done on the relevant Conjur instances?

Reply:
some of the configuration is done on the master (i.e loading the policy with a conjur cli which runs against master) but some of the configuration is run against other conjur instances. For example, enabling the authenticator in the ENV is done in any instance that users will authenticate against (and it's documented).

This is a good point for documentation though, thanks!